### PR TITLE
[MLPerf] Add DLRM-DCNv2

### PR DESCRIPTION
**Pull request by abheesht17**
TODOs:

- [x] Need to make it work for multi-host: https://github.com/keras-team/keras-rs/issues/143.
- [x]  Currently uses dummy data, need to shift it to using the actual dataset. Already added code for loading actual data, but the path we are reading already has "batched" data. Can go ahead with this, or can load "unbatched" data and batch that up.
- [x] Eval dataset
- [x] Check if random seed is being correctly passed. We need to make sure that the initialiser returns different values for different layers (since we are passing `SeedGenerator`, I think we are good, but good to verify manually).
- [ ] ~Instead of separate files for configs, maybe have separate functions for configs.~
- [ ] Is this a worry?

```
 08:44:12.046114: W jax_tpu_embedding/sparsecore/lib/core/input_preprocessing_util.cc:251] No Coo Buffer Size provided for table cat_14_table_cat_15_table_cat_23_table_cat_24_table_cat_25_table_cat_33_table_cat_34_table_cat_35_table_cat_36_table, the default value (6144) may be too large and can cause OOM. Utilize the stats returned from the sparse dense matmul preprocessing API.
```
- [ ] Set up metric logging and checkpointing.

### CODE REVIEW

File: examples/ml_perf/main.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+        )
+        vocabulary_size = large_emb_feature["vocabulary_size"]
+        multi_hot_size = large_emb_feature["multi_hot_size"]
+
+        table_config = keras_rs.layers.TableConfig(
+            name=f"{feature_name}_table",
+            vocabulary_size=vocabulary_size,
+            embedding_dim=embedding_dim,
+            # TODO(abheesht): Verify.
+            initializer=keras.initializers.VarianceScaling(
```
Comment (hertschuh) [inline examples/ml_perf/main.py:57]: The default is `keras.initializers.VarianceScaling(mode="fan_out")`, which has a scale of 1 and a uniform distribution.

Why did you need to change it?

File: examples/ml_perf/main.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+os.environ["KERAS_BACKEND"] = "jax"
+
+import keras
+
+import keras_rs
+
+from .dataloader import DataLoader
+from .model import DLRMDCNV2
+
+SEED = 1337
```
Comment (hertschuh) [inline examples/ml_perf/main.py:14]: It looks like you carefully passed the SEED everywhere, but it's hard to say if you missed one.

I would also do `keras.config.set_random_seed(SEED)` just to make sure things are consistent across hosts.

File: examples/ml_perf/main.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+                mode="fan_in",
+                distribution="uniform",
+                seed=SEED,
+            ),
+            optimizer=keras.optimizers.Adagrad(
+                learning_rate=embedding_learning_rate
+            ),
+            combiner="sum",
+            placement="sparsecore",
+            # TODO: These two args are not getting passed down to
```
Comment (hertschuh) [inline examples/ml_perf/main.py:68]: You could try these XLA flags:
- `xla_sparse_core_max_ids_per_partition_per_sample`
- `xla_sparse_core_max_unique_ids_per_partition_per_sample`

File: examples/ml_perf/main.py [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+            # `jax-tpu-embedding` properly, seems like.
+            max_ids_per_partition=max_ids_per_partition,
+            max_unique_ids_per_partition=max_unique_ids_per_partition,
+        )
+        feature_configs[f"{feature_name}_id"] = keras_rs.layers.FeatureConfig(
+            name=feature_name.replace("id", ""),
+            table=table_config,
+            # TODO: Verify whether it should be `(bsz, 1)` or
+            # `(bsz, multi_hot_size)`.
+            input_shape=(per_host_batch_size, multi_hot_size),
```
Comment (hertschuh) [inline examples/ml_perf/main.py:78]: multi_hot as an input to embedding tables is not supported, it should be turned into a ragged or padded sequence of indices before being fed to the DistributedEmbedding.

Comment (abheesht17) [reply examples/ml_perf/main.py:80]: > multi_hot as an input to embedding tables is not supported, it should be turned into a ragged or padded sequence of indices before being fed to the DistributedEmbedding.

Oh, I think our understanding of multi-hot is different. I am passing indices here. I think what you mean by multi-hot is `[[1, 0, 1, 0, 0], [0, 0, 1, 1, 1]]`

**Review by hertschuh — COMMENTED** [review]
Some comments about the bash script
